File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
  var answer = [];
  let count = 0;
  let zero_count = 0;
  for (lotto_num of lottos) {
    if (lotto_num === 0) {
      zero_count++;
      continue;
    }
    if (win_nums.includes(lotto_num)) count++;
  }
  answer.push([6, 6, 5, 4, 3, 2, 1][zero_count + count]);
  answer.push([6, 6, 5, 4, 3, 2, 1][count]);

  return answer;
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.Set;
import java.util.HashSet;
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int[] rank = new int[] {6,6,5,4,3,2,1};
        HashSet<Integer> hs = new HashSet<>();
        for(int num : win_nums){
            hs.add(num);
        }
        
        int zeroCount = 0;
        for(int i = 0; i < lottos.length; i++){
            if(lottos[i] == 0){
                zeroCount++;
            }else if(hs.contains(lottos[i])){
                answer[1]++;
            } 
        }
    
        answer[0] = answer[1] + zeroCount;
        answer[0] = rank[answer[0]];
        answer[1] = rank[answer[1]];
        return answer;
    }
}